toolkit/tools/internal/file/file.go

```diff
@@ -39,6 +39,17 @@ func IsFile(path string) (isFile bool, err error) {
 	return !info.IsDir(), nil
 }
 
+// IsFileOrSymlink returns true if the provided path is a file or a symlink.
+func IsFileOrSymlink(path string) (isFile bool, err error) {
```

Comment:
nit: since we already have `isFile` could you restrict this function to just `isSymlink` and use both in your usecase?

Reply:
That would mean 2 syscalls instead of 1, which seems a little wasteful.